src/taipy/core/_repository/_sql_repository.py

```diff
@@ -129,8 +129,8 @@ def _search(self, attribute: str, value: Any, version_number: Optional[str] = No
         return self.__to_entity(entry)
 
     def _get_by_config_and_owner_id(self, config_id: str, owner_id: Optional[str]) -> Optional[Entity]:
-        entities = iter(self.__get_entities_by_config_and_owner(config_id, owner_id))
-        return self.__to_entity(next(entities, None))
+        entity = self.__get_entities_by_config_and_owner(config_id, owner_id)
```

Comment:
**Note to reviewers** I _believe_ this code has the same behavior as before (using an `iter` and then a `next`). Let me know if it is not the case...